pkg/platform/kube/test/platform_test.go

```diff
@@ -0,0 +1,195 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"encoding/base64"
+	"testing"
+
+	"github.com/nuclio/nuclio/pkg/functionconfig"
+	"github.com/nuclio/nuclio/pkg/platform"
+	"github.com/nuclio/nuclio/pkg/platform/kube"
+	nuclioio "github.com/nuclio/nuclio/pkg/platform/kube/apis/nuclio.io/v1beta1"
+	"github.com/nuclio/nuclio/pkg/platformconfig"
+
+	"github.com/stretchr/testify/suite"
+	appsv1 "k8s.io/api/apps/v1"
+	autoscalingv1 "k8s.io/api/autoscaling/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type DeployFunctionTestSuite struct {
+	KubeTestSuite
+}
+
+func (suite *DeployFunctionTestSuite) SetupSuite() {
+	suite.KubeTestSuite.SetupSuite()
+
+	// start controller in background
+	go suite.Controller.Start() // nolint: errcheck
+}
+
+func (suite *DeployFunctionTestSuite) TestStaleResourceVersion() {
+	var resourceVersion string
+
+	createFunctionOptions := suite.compileCreateFunctionOptions("resource-schema")
+
+	afterFirstDeploy := func(deployResult *platform.CreateFunctionResult) bool {
+
+		// get the function
+		functions, err := suite.Platform.GetFunctions(&platform.GetFunctionsOptions{
+			Name:      createFunctionOptions.FunctionConfig.Meta.Name,
+			Namespace: createFunctionOptions.FunctionConfig.Meta.Namespace,
+		})
+		suite.Require().NoError(err)
+		function := functions[0]
+
+		// save resource version
+		resourceVersion = function.GetConfig().Meta.ResourceVersion
+		suite.Require().NotEmpty(resourceVersion)
+
+		// ensure using newest resource version on second deploy
+		createFunctionOptions.FunctionConfig.Meta.ResourceVersion = resourceVersion
+
+		// change source code
+		createFunctionOptions.FunctionConfig.Spec.Build.FunctionSourceCode = base64.StdEncoding.EncodeToString([]byte(`
+def handler(context, event):
+  return "darn it!"
+`))
+		return true
+	}
+
+	afterSecondDeploy := func(deployResult *platform.CreateFunctionResult) bool {
+
+		// get the function
+		functions, err := suite.Platform.GetFunctions(&platform.GetFunctionsOptions{
+			Name:      createFunctionOptions.FunctionConfig.Meta.Name,
+			Namespace: createFunctionOptions.FunctionConfig.Meta.Namespace,
+		})
+		suite.Require().NoError(err, "Failed to get functions")
+
+		function := functions[0]
+		suite.Require().NotEqual(resourceVersion,
+			function.GetConfig().Meta.ResourceVersion,
+			"Resource version should be changed between deployments")
+
+		// we expect a failure due to a stale resource version
+		suite.DeployFunctionExpectError(createFunctionOptions, func(deployResult *platform.CreateFunctionResult) bool {
+			suite.Require().Nil(deployResult, "Deployment results is nil when creation failed")
+			return true
+		})
```

Comment:
Move this outside to line 101, it's very confusing written in the callback. These callbacks should contain actions / verifications related to that deployment